design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,327 @@
+  feature and follow it to graduation (GA).
+- As a KubeVirt contributor, I would like to remove a feature
+  that has not yet reached its formal graduation.
+- As a KubeVirt contributor, I would like to remove a feature
+  that has already graduated.
+  > **Note**: Removal of a GA feature is considered an exception.
+  > Strong arguments and a wide agreement is required for such an action.
+- As a KubeVirt cluster operator, I would like to experiment with
+  a newly-proposed ("Alpha") feature in a controlled environment, to see
+  if it makes sense to me.
+- As a KubeVirt cluster operator, I would like to evaluate an
+  undergraduate ("Beta") feature in a real-life environment with actual users.
+- As a KubeVirt cluster operator, I would like to keep using a feature
+  that got graduated after I used it during the "Beta" evaluation period.
+- As a KubeVirt cluster operator, I would like to know that
+  an experimental ("Alpha") feature is planned to be removed.
+- As a KubeVirt cluster operator, I would like to know that
+  an undergraduate ("Beta") feature is planned to be removed.
+- As a KubeVirt cluster operator, I would like to stop using a feature
+  that got removed.
+
+### Repos
+This is a cross repo project policy under the
+[kubevirt](https://github.com/kubevirt) organization.
+
+## Proposal Design
+The proposal on how to define a feature lifecycle is influenced by
+processes and policies from the Kubernetes project.
+These sources are scattered around, each focusing on different
+aspects of a feature:
+- [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+- [Graduation](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/)
+- [Changing the API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+- [Deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+The proposal takes the top-down approach, starting with the high level
+flow that a common feature will traverse through.
+Continuing with actions that need to be taken and timeline suggestions.
+
+Both feature graduation and discontinuation flows are covered.
+Including the implications on users.
+
+Depending on individual topics, follow-up proposals may extend the basic
+points raised in this proposal.
+
+### Terminology
+Feature Gates and feature configuration are often used interchangeable or not differentiated.
+- Feature Gate: A flag that controls the presence or availability of a feature in the cluster.
+- Feature configuration: Cluster or workload level configuration that allows an admin or user
+  (depending on the feature) to control aspects of a feature operation.
+  A common usage is to determine if features are opt-in or opt-out by default.
+
+### Feature Stages
+A feature is expected to pass in the following order through the following stages:
+1. Enhancement proposal.
```

Comment:
Even with this change, I still need to convince two people to accept my changes. Shouldn't we include more reviewers?

Reply:
Such details fit an enhancement guide. E.g. who should review, approve, how many, etc.
It is a different subject.

Reply:
Ok, we can mark this as resolved.